### DISCUSSION TITLE
CRD with 8 thousand lines?

### DISCUSSION THREAD

**Discussion by rodrigoscferraz** [kubernetes-sigs/kubebuilder#2450] [Support] [answered]
Hey folks, just a dumb question

Why the CRD generated by CronJob tutorial has almost 8 thousand lines([here](https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/book/src/cronjob-tutorial/testdata/project/config/crd/bases/batch.tutorial.kubebuilder.io_cronjobs.yaml))? 

Shouldn't it just contain cronjob-related stuff?

thx in advance

**Comment by mogsie** [comment, marked as answer]
It's this big because it embeds a _PodSpec_ — the job to run.  PodSpecs are the bread'n'butter of Kubernetes, and it is common to include a "pod template" in resource types that define workloads.

**Reply by rodrigoscferraz** [reply]
Awesome, thx man!

**Comment by camilamacedo86** [comment]
Closing since it is answered.
